### PR TITLE
[bitnami/wordpress] feat!: :lock: :boom: Improve security defaults

### PR DESCRIPTION
**Pull request by javsalgar**
BREAKING CHANGE

Signed-off-by: Javier Salmeron Garcia <jsalmeron@vmware.com>

### Description of the change

This major bump changes the following security defaults:

- `runAsGroup` is changed from `0` to `1001`
- `readOnlyRootFilesystem` is set to `true`
- `resourcesPreset` is changed from `none` to the minimum size working in our test suites (NOTE: `resourcesPreset` is not meant for production usage, but `resources` adapted to your use case).
- `global.compatibility.openshift.adaptSecurityContext` is changed from `disabled` to `auto`.
- The `networkPolicy` section has been normalized amongst all Bitnami charts. Compared to the previous approach, the values section has been simplified (check the Parameters section) and now it set to `enabled=true` by default. Egress traffic is allowed by default and ingress traffic is allowed by all pods but only to the ports set in `containerPorts` and `extraContainerPorts`.

### Benefits

- Better compliance with NSA and MITRE security checklists
- Improved compatibility with Openshift

### Possible drawbacks

This could potentially break any customization or init scripts used in your deployment. If this is the case, change the default values to the previous ones.

### Applicable issues

- https://github.com/bitnami/charts/issues/24251

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by alemorcuq — APPROVED** [review]

File: bitnami/wordpress/templates/deployment.yaml [hunk 1]
```diff
@@ -107,10 +106,68 @@ spec:
...
+
+              info "Copying base dir to empty dir"
+              # In order to not break the application functionality (such as upgrades or plugins) we need
+              # to make the base directory writable, so we need to copy it to an empty dir volume
+              cp -r --preserve=mode /opt/bitnami/wordpress /bitnami
+              info "Copy operation completed"
+          volumeMounts:
+            - name: empty-dir
+              mountPath: /bitnami
+              subPath: app-base-dir
```
Comment (juan131) [inline bitnami/wordpress/templates/deployment.yaml:137]: why `/bitnami`? Maybe I'm biased as Bitnami developer but that directory has been our traditional persistent directory but here we're using it to hold a copy of the original WordPress source code.

File: bitnami/wordpress/templates/deployment.yaml [hunk 2]
```diff
@@ -107,10 +106,68 @@ spec:
...
+
+              . /opt/bitnami/scripts/libfs.sh
+              . /opt/bitnami/scripts/liblog.sh
+
+              info "Copying symlinks to stdout/stderr"
+              # We copy the logs folder because it has symlinks to stdout and stderr
+              if ! is_dir_empty /opt/bitnami/apache/logs; then
+                cp -r /opt/bitnami/apache/logs /emptydir/apache-logs-dir
+              fi
+              info "Copy operation completed"
```
Comment (juan131) [inline bitnami/wordpress/templates/deployment.yaml:164]: Why don't we do this in the previous init-container?

File: bitnami/wordpress/templates/deployment.yaml (hunk 1 above)
Comment (javsalgar) [reply bitnami/wordpress/templates/deployment.yaml:137]: Changed it

File: bitnami/wordpress/templates/deployment.yaml (hunk 2 above)
Comment (javsalgar) [reply bitnami/wordpress/templates/deployment.yaml:164]: Done

**Review by juan131 — APPROVED** [review]
LGTM

**Pull request merged**